src/components/bottom-bar/styles.ts

```diff
@@ -0,0 +1,48 @@
+import { styled } from '@linaria/react'
+import { ElMenu, ElMenuList, ElMenuPopover } from '../menu'
+
+export const ElBottomBar = styled.div`
+  display: flex;
+  padding: var(--spacing-2, 8px);
+  justify-content: center;
+  align-items: center;
+  align-self: stretch;
+  border-top: var(--border-default, 1px) solid var(--outline-default, #e5e9ed);
+  background: var(--fill-white, #fff);
+
+  position: absolute;
+  bottom: 0;
+  left: 0;
+  z-index: 5;
+  overflow: visible;
+  margin-top: auto;
+  width: 100%;
+
+  transition:
+    transform 0.3s ease-in-out,
+    visibility 0.3s ease-in-out;
+
+  &[data-slide-direction='up'] {
+    transform: translateY(0);
+    visibility: visible;
+  }
+
+  &[data-slide-direction='down'] {
+    transform: translateY(100%);
+    visibility: hidden;
+  }
+
```

Comment:
> Why do we need to tightly couple BottomBar to these Menu classes? This significantly undermines the loose coupling we're aiming for in BottomBar's own interface because these style overrides imply BottomBar only works with these specific Menu classes

interesting issue... overriding the base styles would achieve the same visual outcome from design spec, but it would make the component less flexible with other components. We could decouple `ElMenu` by creating another wrapper that wraps `Menu` but not with `ElMenuList` and `ElMenuPopover`

it seems it would be more beneficial if both components also accepted position control props, so we wouldn't need to tightly couple them, what do you think?

Reply:
I don't understand sorry. What's the issue that you're solving with these style overrides here? Positioning of the menu?

Reply:
>  What's the issue that you're solving with these style overrides here?

To make the trigger should always aligned with another 4 items, without overrides the style we can't achieve how the design should look

here I break them down respectively:
- `ElMenu`
we have the `BottomBarMoreMenu` and it's encapsulate the Menu already, but the it will render the `ElMenu` then it will followed to render the children (`BottomBarItem` in this case), so theoritically the last child of `BottomBar` is `ElMenu` not `BottomBarItem`) so it would not have the same styles that it should have

a preview when we're not override the `ElMenu` style

![{FD4DA9D1-73B5-4FB0-B6EE-DA71ECCD812F}](https://github.com/user-attachments/assets/b2c243f6-adf8-459b-bfb7-f76de967e675)

compare when we override the `ElMenu` style

![{ABDDFE73-B414-41B4-8DFA-90DE82814666}](https://github.com/user-attachments/assets/4ab71156-1d81-42b2-a8ef-2792fea4d5fa)

- `ElMenuList`, change the `min-width` that has been preferred from the design spec; unfortunately the default `ElMenuList` already have `min-width` CSS with `200px`, so I have to override that

- `ElMenuPopover`, amending the Popover position to the right corner side from the entry point; the `yOffset` really helpful there, but unfortunately we don't have the `zOffset` because the `BottomBar` requires the Popover should be right aligned instead center so I have to go with the css approach

Reply:
Okay, thanks for the explanations. Can you please remove all those overrides/workarounds from this PR and we'll deal with them separately. Many of the changes you've described should be things the Menu component facilitates via props; styling other elements from a distance like this (i.e. having some parent like bottom bar override styles in one of its descendants) is a code smell and something we should avoid.

You've correctly identified limitations with the current Menu component, so we need to deal with those on the Menu component.

Reply:
**thought:** agree for `ElMenuList` and `ElMenuPopover` to amend that in `Menu` scope, for `ElMenu` I think it could be done by creating a new style element that wrap the `BottomBarItem` and `Menu`; practically it might only have `flex: 1 0 0` CSS properties and this will be placed in atoms level from `BottomBar` component

```tsx
// bottom-bar.atoms.tsx
export const BottomBarMoreMenu: FC<BottomBarMoreMenuProps> = ({ children }) => {
  return (
     <ElBottomBarChildWrapper>
       // ... rest
    </ElBottomBarChildWrapper>
   )
}

export const BottomBarItem = ({children}) => {
  return (
     <ElBottomBarChildWrapper>
       // ... rest
    </ElBottomBarChildWrapper>
   )
}
```

**note:** we might change the wrapper style name